tests/e2e/tests/test_rds_s3.py

```diff
@@ -124,78 +129,63 @@ def cfn_stack(metadata, cluster, region, request):
             "SecurityGroupId": security_groups[0],
             "DBUsername": rand_name("admin"),
             "DBPassword": rand_name("Kubefl0w"),
+            "S3SecretString": create_secret_string(s3_secret),
         },
     )
 
 
 KFP_MANIFEST_FOLDER = "../../awsconfigs/apps/pipeline"
-KFP_PARAMS_ENV_FILE = KFP_MANIFEST_FOLDER + "/params.env"
+KFP_RDS_PARAMS_ENV_FILE = KFP_MANIFEST_FOLDER + "/rds/params.env"
+KFP_S3_PARAMS_ENV_FILE = KFP_MANIFEST_FOLDER + "/s3/params.env"
 
+AWS_SECRETS_MANAGER_MANIFEST_FOLDER = "../../awsconfigs/common/aws-secrets-manager"
+RDS_SECRET_PROVIDER_CLASS_FILE = AWS_SECRETS_MANAGER_MANIFEST_FOLDER + "/rds/secret-provider.yaml"
+S3_SECRET_PROVIDER_CLASS_FILE = AWS_SECRETS_MANAGER_MANIFEST_FOLDER + "/s3/secret-provider.yaml"
 
-@pytest.fixture(scope="class")
-def rds_s3_secrets(cfn_stack, region, metadata, request):
-    """
-    Fixture to keep track of rds and s3 AWS secret creation.
-    These secrets can't be created as part of the CFN stack because
-    the CFN implementation of secrets manager appends a random string
-    to the end of a secret name. Our current integration with secrets
-    manager has hardcoded the secret names as 'rds-secret' and 's3-secret'
-    so having a random string appended to the end of the secret name
-    would not work with our current implementation.
-
-    Todo: allow secret names to be configurable parameters
-    """
 
+@pytest.fixture(scope="class")
+def configure_manifests(cfn_stack, aws_secrets_driver, region):
     cfn_client = get_cfn_client(region)
     stack_outputs = get_stack_outputs(cfn_client, cfn_stack["stack_name"])
 
-    rds_secret = {
-        "username": cfn_stack["params"]["DBUsername"],
-        "password": cfn_stack["params"]["DBPassword"],
-        "database": "kubeflow",
-        "host": stack_outputs["RDSEndpoint"],
-        "port": "3306",
-    }
-
-    s3_secret = {
-        "accesskey": get_accesskey(request),
-        "secretkey": get_secretkey(request),
-    }
-
-    def on_create():
-        secrets_manager_client = get_secrets_manager_client(region)
-        secrets_manager_client.create_secret(
-            Name="rds-secret-2",
-            SecretString=create_secret_string(rds_secret),
-        )
-        secrets_manager_client.create_secret(
-            Name="s3-secret-2",
-            SecretString=create_secret_string(s3_secret),
-        )
-
-    def on_delete():
-        secrets_manager_client = get_secrets_manager_client(region)
-        secrets_manager_client.delete_secret(
-            SecretId="rds-secret-2", ForceDeleteWithoutRecovery=True
-        )
-        secrets_manager_client.delete_secret(
-            SecretId="s3-secret-2", ForceDeleteWithoutRecovery=True
-        )
+    rds_secret_provider = unmarshal_yaml(RDS_SECRET_PROVIDER_CLASS_FILE)
+    rds_secret_provider_objects = yaml.safe_load(
+        rds_secret_provider["spec"]["parameters"]["objects"]
+    )
+    rds_secret_provider_objects[0]["objectName"] = "-".join(
```

Comment:
to confirm, only one line needs to change for configuring secret name, correct?

Reply:
yes